lib/remix/worker.ex

```diff
@@ -34,7 +32,13 @@ defmodule RemixedRemix.Worker do
   def handle_path(path, current_mtime, current_mtime), do: {path, current_mtime}
 
   def handle_path(path, current_mtime, _) do
-    comp_elixir = fn -> Mix.Tasks.Compile.Elixir.run(["--ignore-module-conflict"]) end
+    comp_elixir =
+      if Mix.Project.umbrella?() do
+        fn -> Mix.ProjectStack.recur(fn -> recursive_compile() end) end
```

Comment:
What is the `Mix.ProjectStack` and what exactly does the `recur` function do here? I'm not familiar with it and I see no documentation on the module on Hexdocs or in the mix source code...

Reply:
Yeah, it is not well documented but it prevents recursion race conditions from parallel requests,  from what i could gather from reading the code. It is a simple Agent that tracks what is being scheduled to run to prevent multiple concurrent calls to the same task